jni_gen/java/src/main/java/com/github/dart_lang/jni_gen/apisummarizer/doclet/ElementBuilders.java

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+package com.github.dart_lang.jni_gen.apisummarizer.doclet;
+
+import com.github.dart_lang.jni_gen.apisummarizer.elements.*;
+import com.github.dart_lang.jni_gen.apisummarizer.util.StreamUtil;
+import com.sun.source.doctree.DocCommentTree;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.lang.model.element.*;
+import javax.lang.model.type.*;
+
+public class ElementBuilders {
+  AstEnv env;
+
+  public ElementBuilders(AstEnv env) {
+    this.env = env;
+  }
+
+  private void fillInFromTypeElement(TypeElement e, ClassDecl c) {
+    c.modifiers = e.getModifiers().stream().map(Modifier::toString).collect(Collectors.toSet());
+    switch (e.getKind()) {
+      case INTERFACE:
+        c.declKind = DeclKind.INTERFACE;
+        break;
+      case CLASS:
+        c.declKind = DeclKind.CLASS;
+        break;
+      case ENUM:
+        c.declKind = DeclKind.ENUM;
+        break;
+      case ANNOTATION_TYPE:
+        c.declKind = DeclKind.ANNOTATION_TYPE;
+        break;
+    }
```

Comment:
Consider adding a default case with throwing an exception, otherwise c.declKind will be uninitialized.